### PR TITLE
feat(ckconv): block-diagonal multi-ω₀ SIREN kernel + aligned Gaussian mask

### PR DESCRIPTION
**Pull request by dwromero**
## Summary

Adds two new CKConv building blocks that broaden the spectral coverage of SIREN-parameterized convolutional kernels at initialization, plus three new ViT-5 hybrid leaf configs that use them, plus the full study that motivated the design.

- **`BlockDiagonalMultiOmegaSIRENKernelND`** (`nvsubquadratic/modules/kernels_nd.py`) — per-block ω₀ schedule on the first SIREN layer + block-diagonal hidden+output linear structure. `K=8` blocks, `off_block_scale=0.1` controls inter-block leakage. Default schedule: `ω₀ ∈ linspace(1, 12, 8)`.
- **`BlockAlignedGaussianModulationND`** (`nvsubquadratic/modules/masks_nd.py`) — variant of `GaussianModulationND` that flips `std_param` along the channel axis so the widest spectral mask is aligned with the lowest-ω₀ block (and the narrowest with the highest-ω₀ block), preventing the default mask from double-suppressing the low band and leaking the high band.
- **3 new ViT-5 leaf configs** (`hybrid_ha_blockdiag.py`, `hybrid_hhha_blockdiag.py`, `full_hyena_blockdiag.py`) that use these kernels as **post-creation overrides** on top of the existing `_base_config` (no changes to the base). The override logic is factored into a small helper module (`_blockdiag.py`).
- **Tests** (`tests/modules/test_multi_omega_siren.py`) — 36 tests covering per-row ω₀ embedding, block-mask shape, strict block-diagonal at `off=0`, schedule sanity (linear/log), and mask-alignment correctness.
- **Reports** (`reports/`) — new top-level `reports/` tree with an indexing README; `ckconv_block_diagonal_kernel/REPORT.md` plus the scripts and figures backing every claim:
  - Variant screen at base resolution (7-panel histogram).
  - `(off_block_scale, ω₀_max)` aliasing-aware grid at base + 2× resolution.
  - 2-D power-spectrum visualizations.
  - Resolution-scaling rule (\"double `m` → double `ω₀_max`\") on both the scalar SIREN baseline and the block-diagonal variant.
  - Production-vs-prototype matching check (the prototype used for the report and the production class produce identical spectra).
- `pyproject.toml`: adds `reports/**/*.py` to ruff per-file-ignores (mirrors the existing `scripts/` + `benchmarks/` treatment for analysis scripts not meant as production APIs).

Production defaults baked into `_blockdiag.py`:

```
num_blocks=8
omega_0_min=1.0
omega_0_max=12.0
schedule="linear"
off_block_scale=0.1
```

## Stacking note

This PR currently targets `main` but is built on top of `dwromero/hybrid-vit5`, so its diff includes the 3 commits from that branch as well. They will drop out of the diff once `dwromero/hybrid-vit5` is merged to `main`.

## Results — ImageNet-1k pretraining (ViT-5 small, `H×12`, patch 8)

End-to-end convergence check on the full Hyena (`H×12`) ViT-5-small (~22.34 M params, 38.72 GFLOPs/image) at patch_size=8, 250k iterations, 1.25% warmup, 400 epochs trained:

| Run | Config | Key kernel knobs | test/acc | val/acc_ema (best) | test/loss |
| --- | ------ | ---------------- | -------- | ------------------ | --------- |
| [`xlujiniz`](https://wandb.ai/implicit-long-convs/nvsubquadratic/runs/xlujiniz) | `full_hyena.py` (baseline scalar SIREN) | `ω₀ = 20.0` | 83.59% | 83.50% (83.60%) | 0.6483 |
| [`5kn6nrzs`](https://wandb.ai/implicit-long-convs/nvsubquadratic/runs/5kn6nrzs) | `full_hyena_blockdiag.py` *(this PR — fixed per-block ω₀)* | `ω₀ ∈ [1, 24]`, `K=8` | 83.38% | 83.26% (83.39%) | 0.6581 |
| [`d2t0692n`](https://wandb.ai/implicit-long-convs/nvsubquadratic/runs/d2t0692n) | `full_hyena_learnable_omega_blockdiag.py` *(learnable ω₀ + block-diag)* | `ω₀ ∈ [1, 24]`, `K=8`, learnable | **83.61%** | 83.56% (83.62%) | 0.6495 |
| [`xdu8ox9f`](https://wandb.ai/implicit-long-convs/nvsubquadratic/runs/xdu8ox9f) | same as above with `apply_lr_scale=False` | `ω₀ ∈ [1, 24]`, `K=8`, learnable, no LR scale | 83.42% | 83.35% (83.42%) | 0.6562 |

Takeaways:

- The block-diagonal kernel **with learnable ω₀** (`d2t0692n`) matches and very slightly surpasses the scalar SIREN baseline (83.61% vs 83.59%), confirming convergence is on par.
- The block-diagonal kernel with **fixed** per-block ω₀ (`5kn6nrzs`, the production default flavor in this PR) trails the baseline by ~0.2 pp, suggesting per-block ω₀ benefits from being trainable in practice.
- Disabling LR scaling on the learnable ω₀ (`xdu8ox9f`) recovers most but not all of the gap (83.42%), so the LR-scaling on ω₀ is a meaningful contributor.

## Test plan

- [x] `conda run -n nv-subq python -m pytest tests/modules/test_multi_omega_siren.py` — 36 passed
- [x] `ruff check` clean across all changed files
- [x] `ruff format` clean (pre-commit hook)
- [x] Smoke test: all three new leaf configs (`hybrid_ha_blockdiag`, `hybrid_hhha_blockdiag`, `full_hyena_blockdiag`) instantiate end-to-end and run a forward pass
- [ ] CI green on PR
- [x] Quick training run on one of the new configs to confirm convergence is at least on par with the non-blockdiag baseline — see [Results](#results--imagenet-1k-pretraining-vit-5-small-h12-patch-8) section above (learnable-ω₀ block-diagonal matches baseline; fixed-ω₀ block-diagonal trails by ~0.2 pp)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by Dafidofff — APPROVED** [review]
LGTM
